File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string, vector<string>> mstr;
        for(int i = 0; i < strs.size(); i++){
            string strBefore = strs[i];
            sort(strs[i].begin(), strs[i].end());
            mstr[strs[i]].push_back(strBefore);
        }
        vector<vector<string>> strM;
        for(auto m : mstr){
            strM.push_back(m.second);
        }
        return strM;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashmap = {}
        for word in strs:
            sorted_word = ''.join(sorted(word))
            if sorted_word not in hashmap:
                hashmap[sorted_word] = [word]
            else:
                hashmap[sorted_word].append(word)
        words = []
        for sorted_word in hashmap:
            words.append(hashmap[sorted_word])
        return words
